added extra validation for various inputs

tbuwadi opened the pull request:
added validation for:
- dates in customer new reservation view
- dates in admin new reservation view
- dates in modify reservation view
- cell number in new customer registration view
- credit card number in new customer registration view

changed:
- password field in new customer registration to be *** instead of text

tbuwadi committed:
added extra validation for various inputs

mclhtay commented on HotelBooking/Views/book_reservation_view.py:
```diff
@@ -4,6 +4,15 @@
 from HotelBooking.Models.room import ROOM_TYPE
+from datetime import datetime
+
+def ValidateDate(date_text):
```
can you move this function to Views/utils.py? 
Also, snake case.

mclhtay replied:
Refactor it to Utils, so the same validation method can be shared by all other views.

tbuwadi replied:
yessss

mclhtay commented on HotelBooking/Views/reserved_rooms_admin_view.py:
```diff
@@ -5,6 +5,15 @@
 from PyInquirer import prompt
 from HotelBooking.Models.room import ROOM_TYPE
+from datetime import datetime
+
```
Not a comment but a complaint. 

LMAO who named this file? Why does it have such a weird name? Do we have a `reserved_rooms_customer_view` then?

tbuwadi replied:
lolll didn't you make the admin stuff:')

mclhtay replied:
I'm pretty sure I made a file named `reserved_rooms_view`. Someone else split the difference.

mclhtay replied:
Doesn't matter I guess, leave it be.

tbuwadi committed:
added validatedate function to views/utils

mclhtay commented on HotelBooking/Views/utils.py:
```diff
@@ -13,3 +13,10 @@ def medium_print(text: str):
 def error_print(prefix: str = "", suffix: str = ""):
     print(f'\n{prefix}Please try again{suffix}')
+
+def ValidateDate(date_text):
```
Stop it! `validate_date` pls

tbuwadi replied:
BE PATIENT ADDING ANOTHER COMMIT RN

mclhtay commented on HotelBooking/Views/utils.py:
```diff
@@ -13,3 +13,10 @@ def medium_print(text: str):
+
+def ValidateDate(date_text):
+        try:
+            datetime.strptime(date_text, '%m/%d/%Y')
```
What's the rationale behind testing it as mm/dd/yy? Checking the SQL file it appears we have a different date format?

https://github.com/mclhtay/CS4471/blob/main/HotelBooking/Database/create_database.sql#L89

mclhtay replied:
Perhaps change the date format in the SQL since you already put mm/dd/yy in a bunch of places.

tbuwadi committed:
snake case smh

tbuwadi committed:
changed date format everywhere to be consistent with mm/dd/yyyy

tbuwadi commented:
@mclhtay 
added snake case and changed dates in sql and tests to be consistent.. didn't necessarily use same dummy dates but made sure all tests pass

mclhtay reviewed (approved):
Approve on the basis that the author did manual UI testing.
